SignalMessaging/attachments/MessageApprovalViewController.swift

```diff
@@ -23,13 +23,7 @@ public class MessageApprovalViewController: OWSViewController, UITextViewDelegat
     let contactsManager: OWSContactsManager
 
     private(set) var textView: UITextView!
-    private var sendButton: UIBarButtonItem = {
-        return UIBarButtonItem(title: NSLocalizedString("SEND_BUTTON_TITLE",
-                                                        comment: "Label for the send button in the conversation view."),
-                               style: .plain,
-                               target: self,
-                               action: #selector(sendPressed))
-    }()
+    private var sendButton: UIBarButtonItem!
```

Comment:
TIL that if you access `self` in a block used to initialize a class property, it has an invalid value.  Send button was broken.

@michaelkirk Is there anywhere else we did something similar?

Reply:
I'm OK with the change, but I don't understand what was wrong with the existing code.

> initialize a class property

I'm a little confused by terminology here - Swift doesn't have "class" properties the way it has "class" functions, it has instance properties and static properties. I think `static` properties are pretty close to what you might call a `class` property, but this is an instance property not a static property.

If you are saying that you *can't* reference `self` inside a block used to initialize a lazy var, I don't think that's correct (unless you are trying to use the lazy var in the initializer, in which case standard "you can't reference self until initialization is complete" holds).

Reply:
Sorry, I misspoke - I should have said _instance_ property. 

But the issue here is that the block used to initialize the property is defined in the context of the class, not the instance.  `self` as it was referred to the `MessageApprovalViewController` class, as it would in a class method.  I was surprised too.